### PR TITLE
Add `from json(string): Any` conversion function

### LINKED ISSUE #825: Add function to parse an escaped JSON string
**Is your feature request related to a problem? Please describe.**
When use the REST connector to call a service, i got the response like below.
Now i need to extract the `taskTodoList` into an array to the result expression.

```
{
    "status": 200,
    "headers": {
        "date": "Fri, 12 Apr 2024 01:29:49 GMT",
        "server": "nginx",
        "transfer-encoding": "chunked",
        "x-content-type-options": "nosniff",
        "x-xss-protection": "1; mode=block",
        "x-frame-options": "SAMEORIGIN",
        "content-type": "application/json;charset=utf-8",
        "connection": "keep-alive",
        "strict-transport-security": "max-age=172800; includeSubDomains"
    },
    "body": {
        "RESPONSE": {
            "RETURN_CODE": "S",
            "RETURN_DATA": "{\"taskTodoList\":[{\"activityDefId\":\"obj_ca954d098ba000018cd81cfdc2602910\",\"activityDefName\":\"startTask\",\"activityModelExt\":{\"customUniqueId\":\"\",\"extendProperty\":\"\",\"id\":\"obj_ca954d098ba000018cd81cfdc2602910\",\"name\":\"startTask\",\"no\":1,\"sequential\":-1},\"activityType\":\"userTask\",\"async\":false,\"beginEngineNode\":\"2406:440:600::16e0:10018\",\"beginTime\":1712885389000,\"claimResourceId\":\"\",\"claimType\":0,\"controlState\":\"active\",\"customUniqueId\":\"\",\"delayTimes\":0,\"delegateUser\":\"\",\"dispatchId\":\"4a9cf601-b664-4cda-8581-ead5814ac6a0\",\"eAITask\":false,\"ext1\":\"\",\"ext2\":\"\",\"ext3\":\"\",\"ext4\":\"\",\"ext5\":\"\",\"ext6\":\"\",\"ext7\":0,\"ext8\":0.0,\"historyTask\":false,\"iOBD\":\"\",\"iOC\":\"\",\"iOR\":\"\",\"iOS\":\"\",\"id\":\"2c8eebfa-7f16-4c14-8f70-bd660d65e510\",\"monitor\":false,\"owner\":\"admin\",\"ownerDepartmentId\":\"5bc3a2dc-3bd2-4376-bcc3-5612e28e55fe\",\"ownerName\":\"admin\",\"parentTaskInstId\":\"00000000-0000-0000-0000-000000000000\",\"priority\":1,\"processDefId\":\"obj_1b01dfade2fd4c408f6426ee290743d0\",\"processDefVerId\":\"obj_1b01dfade2fd4c408f6426ee290743d0\",\"processGroupId\":\"obj_f953af04fb5944ca8128e262d363f62c\",\"processInstId\":\"e4710d5c-9078-4a47-8868-4d43805d5d23\",\"readState\":0,\"remindTimes\":0,\"root\":true,\"scopeId\":\"00000000-0000-0000-0000-000000000000\",\"securityLevel\":0,\"state\":1,\"target\":\"admin\",\"targetCompanyId\":\"8911e732-b42a-4556-853f-ad32761bcbee\",\"targetDepartmentId\":\"5bc3a2dc-3bd2-4376-bcc3-5612e28e55fe\",\"targetName\":\"admin\",\"targetRoleId\":\"d102c89d-55f3-4865-9d5c-c00b7f47b803\",\"targetRoleNo\":\"\",\"taskInfo\":\"\",\"title\":\"Test Zeebe Rest connector\",\"trash\":true}],\"processInstId\":\"e4710d5c-9078-4a47-8868-4d43805d5d23\"}",
            "RETURN_DESC": "sucess",
            "RETURN_STAMP": "2024-04-12 09:29:49:022"
        }
    }
}
```

As discuss in the forum:
 [How to get the taskTodoList from the response body](https://forum.camunda.io/t/how-to-get-the-tasktodolist-from-the-response-body/51773)
[Parse a json string to FEEL context](https://forum.camunda.io/t/parse-a-json-string-to-feel-context/44755)

**Describe the solution you'd like**
A new built-in function to handle the JSON string like `JSON.parse()`.

```
// Function signature
from json(json: string): Any

// Examples
// 1) JSON object to FEEL context
from json("{\"a\": 1, \"b\": 2}")
// {a: 1, b: 2}

// 2) JSON literal to FEEL value
from json("1")
// 1

from json("\"a\"")
// "a"

from json("true")
// true

from json("null")
// null

// 3) JSON array to FEEL list
from json("[1, 2, 3]")
// [1, 2, 3]

// 4) JSON string of a date/time/date-time to FEEL string 
from json("\"2023-06-14\"")
// "2023-06-14" 

from json("\"14:55:00\"")
// "14:55:00" 

from json("\"2023-06-14T14:55:00\"")
// "2023-06-14T14:55:00" 

// 5) JSON string of a duration to FEEL string 
from json("\"P1Y\"")
// "P1Y" 

from json("\"PT2H\"")
// "PT2H" 

// 6) invalid JSON to FEEL null 
from json("invalid")
// null 
```

**Related issues**

* https://github.com/camunda/feel-scala/issues/602
* Camunda Autormation Platform 7: 
* Zeebe broker: 
* OMG issues:
  * https://issues.omg.org/browse/DMN16-68
  * https://issues.omg.org/browse/DMN15-102

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds a new `from json(string): Any` built-in conversion function, allowing users to convert stringified-JSON results (very common with agentic orchestration) into objects, such as contexts, lists, strings, etc. This simplifies extracting, transforming, etc., parts from such results.

The tests were largely taken [from this comment](https://github.com/camunda/feel-scala/issues/602#issuecomment-1591197695), major thanks for this :pray: 

One small issue, Jackson requires an additional compiler file, `Ytasty-reader`. I'm a complete Scala noob, so while I could add it and make progress, I cannot speak for its impact and if that's something we want to move forward with: does it mean downstream users also need to add this flag? If so, do we want that?

## Related issues

closes #825

### CODE REVIEW

**Comment by npepinpe** [comment]
Happy to implement #602 if this is merged. I expect the complexity is similar, and these kind of things are all I can really manage nowadays :(

**Review by saig0 — CHANGES_REQUESTED** [review]
@npepinpe thank you for your contribution. :tada: Your Scala skill increased by 3 EXP. :muscle: 

It looks good. I added a few minor comments. :cookie:

File: pom.xml [hunk 1]
```diff
@@ -177,6 +189,9 @@
               <goal>add-source</goal>
               <goal>compile</goal>
             </goals>
+            <configuration>
+              <addScalacArgs>-Ytasty-reader</addScalacArgs>
+            </configuration>
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply pom.xml:194]: See, this is what happens because I don't know the Scala ecosystem at all, and I really just wanted my function to parse JSON :smile:

File: src/main/scala/org/camunda/feel/impl/builtin/ConversionBuiltinFunctions.scala [hunk 2]
```diff
@@ -301,6 +308,21 @@ class ConversionBuiltinFunctions(valueMapper: ValueMapper) {
...
+    params = List("json"),
+    invoke = {
+      case List(ValNull)         => ValNull
+      case List(json: ValString) =>
+        Try(Mapper.readValue(json.value, classOf[Any]))
+          .map { value: Any =>
+            valueMapper.toVal(value)
+          }
+          .getOrElse {
+            ValError(s"Failed to parse JSON from '${json.value}'")
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply src/main/scala/org/camunda/feel/impl/builtin/ConversionBuiltinFunctions.scala:321]: Can you elaborate what you mean by return the error _first_, and then the JSON? Is there an example of something similar? Like, return a tuple?

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala [hunk 3]
```diff
@@ -528,4 +528,54 @@ class BuiltinConversionFunctionsTest
     ) should returnResult(Period.parse("P1Y8M"))
   }
 
+  "A from json(string) function" should "convert a JSON string" in {
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala:531]: The contribution guide is not very clear though:

```
Add a new test case following the pattern: "A reverse() function" should "return a string in reverse order" in { ??? }
Add more test cases for other results, alternative parameters, or covering edge cases
```

But also

```
Group all test cases for a method using it should "return null if the argument is not a string" in { ??? }
```

What is the criteria for splitting test cases? Input? Output? Both?

**Comment by npepinpe** [comment]
Addressed most of the review feedback, thanks! Just wasn't sure about two points:

- The test split. It's still unclear to me, even from the contributing guide, if the split is over input, output, both, or...? The test file itself doesn't seem to always be consistent.
- I didn't get the point about the error return, i.e. return the error _first_, then the JSON? What's the two-step return? :thinking:

**Review by saig0 — APPROVED** [review]
Nice. :+1:  I grouped the test cases.  

It is ready to merge. :rocket:
